gradio/components/dataset.py

```diff
@@ -32,7 +32,7 @@ def __init__(
         component_props: list[dict[str, Any]] | None = None,
         samples: list[list[Any]] | None = None,
         headers: list[str] | None = None,
-        type: Literal["values", "index"] = "values",
+        type: Literal["values", "index", "tuple"] = "values",
```

Comment:
Added this because in order for caching to work properly, I need the index, and in order for regular examples to work, I need the values.